test/test_torch.py

```diff
@@ -6892,8 +6892,12 @@ def message():
             with self.assertRaisesRegex(TypeError, 'cond must be a bool'):
                 check_fn('wrong type')
 
-            with self.assertRaisesRegex(TypeError, 'cond must be a bool'):
-                check_fn(torch.tensor(True))
+        torch._check(torch.tensor(True))
+        with self.assertRaisesRegex(RuntimeError, 'Expected cond to be True'):
```

Comment:
I thought it would support single element tensor, but here you are extending the meaning to tensor all elements?
is there a specific use case you are targettig?